modules/apps/headless/headless-delivery/headless-delivery-test/src/testFunctional/macros/DocumentFolderAPI.macro

```diff
@@ -4,25 +4,72 @@ definition {
 		var portalURL = JSONCompany.getPortalURL();
 
 		if (isSet(assetLibraryId)) {
-			var api = "asset-libraries/${assetLibraryId}";
+			var api = "asset-libraries/${assetLibraryId}/document-folders";
+		}
+		else if (isSet(parentDocumentFolderId)) {
+			var api = "document-folders/${parentDocumentFolderId}/document-folders";
 		}
 		else {
 			var siteId = JSONGroupAPI._getGroupIdByNameNoSelenium(
 				groupName = ${groupName},
 				site = "true");
 
-			var api = "sites/${siteId}";
+			if (isSet(externalReferenceCode)) {
+				var api = "sites/${siteId}/documents-folder/by-external-reference-code/${externalReferenceCode}";
+			}
+			else {
+				var api = "sites/${siteId}/document-folders";
+			}
+		}
+
+		if (isSet(parameter)) {
+			var api = StringUtil.add(${api}, "?${parameter}=${parameterValue}", "");
 		}
 
 		var curl = '''
-			${portalURL}/o/headless-delivery/v1.0/${api}/document-folders \
+			${portalURL}/o/headless-delivery/v1.0/${api} \
 				-u test@liferay.com:test \
 				-H Content-Type: application/json \
 		''';
 
 		return ${curl};
 	}
 
+	macro addDocumentFolder {
+		if (isSet(externalReferenceCode)) {
+			var body = '''"name": "${name}","externalReferenceCode": "${externalReferenceCode}"''';
```

Comment:
we can remove the else since name is required when adding a document folder:
var body = '''name...''';
if(iseSet(erc)){
  var boday ='''name+erc''';
}